eng/tools/Directory.Build.props

```diff
@@ -8,4 +8,11 @@
     <RestoreSources Condition="'$(DotNetBuildSourceOnly)' == 'true' and '$(IsTestRun)' != 'true'">$(PrebuiltPackagesPath);$(PreviouslySourceBuiltPackagesPath)</RestoreSources>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(DotNetBuildSourceOnly)' == 'true'">
```

Comment:
@mthalman - this prevents the PSB artifacts from being used?  Viktor and I were trying to use PSB to reduce maintenance cost.  w/o this, these versions would need to be added to SBRP and then rebootstrap to pick them up.

Reply:
Then the problem is this:

https://github.com/dotnet/dotnet/blob/3a4520155cd43fe9bf469961e91159822ba67405/eng/tools/Directory.Build.props#L5-L9

That specifically prevents PSB feed from being used for the tests. That was done here: https://github.com/dotnet/sdk/pull/46442

Reply:
PSB is only needed during the build and must not be used at test execution time. But the PSB Versions import was applying to testing as well though. I fixed that with the latest commit and the failures are gone now.